source/fswatch.d

```diff
@@ -238,9 +238,11 @@ struct FileWatch
 		import std.string : toStringz, fromStringz;
 		import std.conv : to;
 
-		private int fd, wd;
+		private int fd;
+		private bool[] wds = new bool[1024]; // mark every watch descriptor
 		private ubyte[1024 * 4] eventBuffer; // 4kb buffer for events
 		private pollfd pfd;
+		private string[int] directoryMap; // map every watch descriptor to a directory
 
 		/// Creates an instance using the linux inotify API
 		this(string path, bool ignored1 = false, bool ignored2 = false)

```

Comment:
why didn't you rename the ignored1 parameter to recursive like in the windows version and left the current behaviour in? I didn't notice that recursive watching didn't work properly on linux so I made this ignored1, but if you implement it you should fix the old behaviour and make this optional like in the other implementations instead of forcing it to match the readme.